src/Adapter/AbstractAdapter.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+/*
+ * This file is part of the Tissue library.
+ *
+ * (c) Cas Leentfaar <info@casleentfaar.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace CL\Tissue\Adapter\ClamAv\Adapter;
+
+use CL\Tissue\Adapter\ClamAv\Model\Detection;
+use CL\Tissue\Adapter\ClamAv\Model\ScanResult;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Process\Process;
+
+abstract class AbstractAdapter implements AdapterInterface
+{
+    protected array $options = [];
+
+    private ?OptionsResolver $resolver = null;
+
+    public function scan(array $paths, array $options = []): ScanResult
+    {
+        $this->options = $this->resolveOptions($options);
+
+        return $this->scanArray($paths);
+    }
+
+    protected function scanArray(array $paths): ScanResult
+    {
+        $files = [];
+        $detections = [];
+        $scannedPaths = [];
+
+        foreach ($paths as $path) {
+            $result = $this->scanSingle($path);
+            $scannedPaths[] = $result->getPaths();
```

Comment:
The previous code had some weird interactions with `$paths` it iterates on.
It merged the scanned paths to the input paths, which was hard to read and also would end in (some) duplicate paths at least.

https://github.com/simonberger/tissue/blob/master/Adapter/AbstractAdapter.php#L51-L56